File: backend/src/main/java/z9/hobby/domain/checkin/dto/CheckInRequestDto.java
package z9.hobby.domain.checkin.dto;

import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

public class CheckInRequestDto {
    @Getter
    @Builder
    @AllArgsConstructor
    @NoArgsConstructor
    public static class CheckInDto {
        @NotNull(message = "Schedule ID must not be null")
        private Long scheduleId;

        @NotNull(message = "Check-in status must not be null")
        private Boolean checkIn;
    }
}

File: backend/src/main/java/z9/hobby/domain/checkin/dto/CheckInRequestDto.kt
package z9.hobby.domain.checkin.dto

import jakarta.validation.constraints.NotNull

data class CheckInRequestDto(
    @field:NotNull(message = "Schedule ID must not be null")
    val scheduleId: Long? = null,

    @field:NotNull(message = "Check-in status must not be null")
    val checkIn: Boolean? = null
)
